theano/gpuarray/opt.py

```diff
@@ -570,22 +854,29 @@ def local_gpu_pdbbreakpoint_op(node):
 
 @register_opt('fast_compile')
 @op_lifter([IfElse])
-def local_gpua_lazy_ifelse(node, context_name):
-    if node.op.gpu:
+@register_opt2([IfElse], 'fast_compile')
+def local_gpua_lazy_ifelse(op, context_name, inputs, outputs):
+    if op.gpu:
         return
-    c = node.inputs[0]
+    # this node is already on GPU, so don't change the graph
+    if isinstance(inputs[0].type, GpuArrayType):
+        return
```

Comment:
Here, the tests fail if i remove that condition. I'm examining why that would happen.  Probably because the Op is same for CPU and GPU here, except that they vary by a parameter ?